docs/docs/query-filters.md

```diff
@@ -0,0 +1,171 @@
+
+In the next guide, a "node with partial path" is any node that has none, or only a part, of the filter path. Internally that results in `undefined`, as any non-existing property would be. For brevity there's no distinction between a partial and non-existing path.
+
+Specific rules:
+
+- `eq`
+  - This is like `filterValue === resultValue` in JavaScript
+  - Can only be used with numbers, strings, booleans, and null
+  - Strict comparison except for `null`
+  - When matching `null`, and only then, it also returns all nodes with partial paths
+- `ne`
+  - This is like `filterValue !== resultValue` in JavaScript
+  - Can only be used with numbers, strings, booleans, and null
+  - Strict comparison except for `null`
+  - Will always return any nodes with partial paths
+- `in`
+  - This is like `filterValue.includes(resultValue)` in JavaScript
+  - Can only be used with an array of numbers, strings, booleans, and null
+  - Strict comparison
+  - When matching `null`, and only then, it also returns all nodes with partial paths
+- `nin`
+  - This is like `!filterValue.includes(resultValue)` in JavaScript
+  - Can only be used with an array of numbers, strings, booleans, and null
+  - Strict comparison
+  - Will always return any nodes with partial paths
+- `lt`, `lte`, `gt`, `gte`
+  - This is like `<`, `<=`, `>`, `>=` respectively
+  - Can only be used with numbers, strings, booleans, and null
+  - If filter value is `null`;
+    - `lt` and `gt` will never match anything
+    - `lte` and `gte` will only match nodes with a result value of exactly `null`
+  - Will never return nodes with partial paths
+  - Weak comparison due to the nature of the JavaScript operators
+- `regex`, `glob`
+  - This is like `new RegExp(filterValue).test(resultValue)` (with caveats for the filterValue syntax)
+  - Glob pattern is converted to a JavaScript RegExp with [micromatch](https://github.com/micromatch/micromatch)
+  - The `regex` filterValue must be a stringified regular expression, including leading and trailing forward slash and optional flags; Like `"/foo/g"`
+  - Never returns nodes with partial paths
+  - While testing, result values are explicitly casted to a string through `String(resultValue)` before passing it to `regex.test()`
+
+### Nulls and partial paths
+
+Due to legacy support for MongoDB compatibility, there are edge cases for each comparator when it comes to `null` values and for partial or non-existing paths. It's best to try and avoid these cases altogether.
+
+## Performance
+
+The key metric that impacts the performance of your queries is the node count for your type of nodes. You can see a dump of these counts when using the `--verbose` flag while building (`gatsby build --verbose`). It tells you the node counts per type during bootstrap. There will be a separate message of the number of page nodes, since they are generated later.
```

Comment:
```suggestion
The key metric that impacts the performance of your queries is the node count for your type of nodes. You can see a dump of these counts when using the `--verbose` flag while building (`gatsby build --verbose`). It tells you the node counts per type during the bootstrap sequence. There will be a separate message of the number of page nodes, since they are generated later.
```